### DISCUSSION TITLE
End Effector Attachment and Program Failure

### DISCUSSION THREAD

**Discussion by dbenner63** [visose/Robots#224] [Q&A] [answered]
Hello! Looking to get some help on end effectors. I have a few mesh end effectors but when  i attach them to the robot they "float in space"

Additionally, when I add a different end effector it makes the program component error out. 

Please help with some suggestions!

**Comment by robin-gdwl** [comment]
could you upload a grasshopper file with the meshes internalised? 
The mesh needs to be positioned on the origin in Rhino (0,0,0) otherwise it will not sit directly on the robot flange. 
What error are you getting on the program component?

**Comment by visose** [comment]
@dbenner63: https://en.wikipedia.org/wiki/Minimal_reproducible_example

**Comment by visose** [comment, marked as answer]
Closing this as stale. For the floating end-effector issue, the tool mesh should be modeled relative to the robot flange/origin before creating the Tool; otherwise it will appear offset in simulation. The program failure needs a minimal Grasshopper file with internalized meshes and the exact error to diagnose. Please reopen with that file if it still reproduces.
